.github/workflows/go.yaml

```diff
@@ -0,0 +1,49 @@
+name: Go
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  basic-checks:
+    name: Run basic-checks
+    runs-on: ubuntu-latest
```

Comment:
I would think we'd want to use docker to run the checks and tests, also to avoid having to `setup-go@v2` every time. Is there a way to do this?

I think this describes how? https://help.github.com/en/actions/creating-actions/creating-a-docker-container-action#creating-an-action-metadata-file Could we then point it to our `Dockerfile.testing` file, then run all our `go vet/fmt/test` commands in that using `uses: actions/our-name`?

Reply:
also we should use the same image that's use in the container e.g. buster-slim https://github.com/docker-library/openjdk/blob/bce2fa373dc270cccf539a8e31b5f2a432d23738/11/jre/slim/Dockerfile#L1

Reply:
👍